### ISSUE TITLE
Add focusTrap attribute to Calendar

### ISSUE THREAD

**Issue by Jonas-Marty** [primefaces/primeng#8968]
**I'm submitting a ...**  (check one with "x")
```
[ x ] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**
With current version 9.1 it is no longer possible to navigate to the next input by tab.
https://stackblitz.io/edit/github-uqnbfh

In 8.x this has been possible
https://www.primefaces.org/primeng/v8.2.5-lts/#/calendar

**Current behavior**
When calendar input gets focused, focus is trapped in the input and cycling over the elements of the calendar overlay. 

**Expected behavior**
I want to navigate to the next element by tab, not to any element of the calendar overlay. If the change to trap focus was made with intention then this issue is more like a feature request to additionally support the old behavior

**Minimal reproduction of the problem with instructions**
https://stackblitz.io/edit/github-uqnbfh

**What is the motivation / use case for changing the behavior?**
Our application is designed as keyboard input first, the calendar overlay is there only as an optional component. More specifically we use it with timeOnly. It is placed inside a table were the user enters data by navigating to the next column by tab.

**Please tell us about your environment:**
* Windows 10 for dev
* IIS for production
* Visual Studio/VS code
* NPM
* **Angular version:** 9.1.4
* **PrimeNG version:** 9.1.0
* **Browser:** Chrome 83 | Firefox  ESR 77 
* **Language:** TypeScript 3.7.5
* **Node (for AoT issues):** `node --version` =   12.18.0

**Comment by Jonas-Marty** [comment]
I created a workaround by hooking into the `keyDown` event and do the navigation myself. Everytime the calendar gets focused it checks if it already attached a `keyDown` event and if not creates one with the provided element ids.

`component.html`
```html
<p-calendar #cal ... (onFocus)="onFocus(cal, rowData)"></p-calendar>
```
`component.ts`
```typescript
onFocus(cal: Calendar, rowData: MyCustomDataType): void {
    CalendarUtils.configureTabNaviation(cal, "previous-element-" + rowData.Id, "next-element-" + rowData.Id);
  }
```
`calendarUtils.ts`
```typescript
static configureTabNaviation(cal: Calendar, backElementId: string, furtherElementId: string): void {
    const attributeName = "tab-navigation-configured";
    let input = document.getElementById(cal.inputId);
    if (!input.hasAttribute(attributeName)) {
      input.addEventListener("keydown", e => {
        if (e.keyCode === 9) {
          let elementToFocus = e.shiftKey
            ? document.getElementById(backElementId)
            : document.getElementById(furtherElementId);
          if (!elementToFocus) {
            elementToFocus = input; //focus self to overwrite alternative behavior of calendar
          }
          e.preventDefault();
          elementToFocus.focus();
          if (elementToFocus instanceof HTMLInputElement && elementToFocus.getAttribute("type") === "text") {
            (elementToFocus as HTMLInputElement).select();
          } else if (elementToFocus instanceof HTMLTextAreaElement) {
            (elementToFocus as HTMLTextAreaElement).select();
          }
        }
      });
      input.toggleAttribute(attributeName, true);
    }
  }
```

**Comment by digitalcraftco** [comment]
Yeah, I really need a solution for this. I also have a Keyboard first POS app where users have been trained for speed and keyboard use. I think the changed happen in v9 to make it more accessible which is awesome but having a way to disable the focus trap without ViewChild would be great.

**Comment by Jonas-Marty** [comment]
Thanks for the fast implementation in 9.1.2, but the behavior is still not what I am expecting. If I set focusTrap="false", and set the focus manually to the calendar input and press <TAB> the overlay still catches focus rather then focusing the next input field (at least the overly does not trap the focx anymore).
Interestingly if I press <SHIFT> + <TAB> I get the desired behavior, the focus jumps directly to the previous input.

I updated the StackBlitz to use v9.1.2 https://stackblitz.com/edit/github-uqnbfh-6qdzys

Tested on latest Chrome and Firefox
